nixos/modules/services/continuous-integration/woodpecker/server.nix

```diff
@@ -0,0 +1,82 @@
+{
+  config
+  , lib
+  , pkgs
+  , ...
+}:
+
+let
+  cfg = config.services.woodpecker-server;
+in {
+  meta.maintainers = [ lib.maintainers.janik ];
+
+
+  options = {
+    services.woodpecker-server = {
+      enable = lib.mkEnableOption (lib.mdDoc "Woodpecker-Server a CI/CD application for automatic builds, deployments and tests.");
+      package = lib.mkPackageOptionMD pkgs "woodpecker-server" { };
+      stateDir = lib.mkOption {
+        default = "/var/lib/woodpecker";
+        type = lib.types.str;
+        description = lib.mdDoc "woodpecker data directory.";
+      };
+
+      environment = lib.mkOption {
+        default = {};
+        type = lib.types.attrsOf lib.types.str;
+        example = lib.literalExpression
+          ''
+            {
+              WOODPECKER_AGENT_SECRET = "your-shared-secret-goes-here";
+              WOODPECKER_HOST = "https://woodpecker.example.tld";
+              WOODPECKER_OPEN = "true";
+              WOODPECKER_GITEA = "true";
+              WOODPECKER_GITEA_CLIENT = "ffffffff-ffff-ffff-ffff-ffffffffffff";
+              WOODPECKER_GITEA_SECRET = "gto_**************************************";
+              WOODPECKER_GITEA_URL = "https://git.example.tld";
+            }
+          '';
+        description = lib.mdDoc "woodpecker config envrionment variables";
+      };
+      user = lib.mkOption {
+        type = lib.types.str;
+        default = "woodpecker";
+        description = lib.mdDoc "service user.";
```

Comment:
thats a bit short

Reply:
I used most of the gitea service as template, they also only use two words, if you have a better suggestion would love to hear that. I will capitalize the first word.